android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/runtime/MapControls.kt

```diff
@@ -0,0 +1,90 @@
+import androidx.compose.foundation.layout.systemBars
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.produceState
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.maplibre.compose.runtime.localLayoutDirection
+import com.maplibre.compose.settings.AttributionSettings
+import com.maplibre.compose.settings.CompassSettings
+import com.maplibre.compose.settings.LogoSettings
+import com.maplibre.compose.settings.MapControlPosition
+import com.maplibre.compose.settings.MapControls
+import com.stadiamaps.ferrostar.composeui.runtime.paddingForGridView
+
+/**
+ * Returns the map controls for the given configuration.
+ *
+ * @param arrivalViewHeight The height of the arrival view.
+ * @param horizontalPadding The horizontal padding to apply to the map controls. Defaults to 16.dp
+ *   to match the default padding of the arrival view.
+ * @param verticalPadding The vertical padding to apply to the map controls from the top of the
+ *   arrival view. Defaults to 8.dp.
+ *
+ * TODO: This function is attempting to optimize the map controls for many screen sizes, system
+ *   bars, and orientations. We should remain open to feedback for specific cases.
+ */
+@Composable
+internal fun rememberMapControlsForArrivalViewHeight(
+    arrivalViewHeight: Dp = 0.dp,
+    horizontalPadding: Dp = 16.dp,
+    verticalPadding: Dp = 8.dp
+): State<MapControls> {
+  val layoutDirection = localLayoutDirection()
+  val density = LocalDensity.current
+  val isLandscape = LocalConfiguration.current.orientation == Configuration.ORIENTATION_LANDSCAPE
+
+  val gridPadding = paddingForGridView()
+  val windowInsetPadding = WindowInsets.systemBars.asPaddingValues()
+
+  return produceState(
+      initialValue = MapControls(),
+      key1 = arrivalViewHeight,
+      key2 = gridPadding,
+      key3 = windowInsetPadding) {
+        // This calculation clamps the controls to the trailing edge of the screen in landscape mode
+        // with less padding in that case. The reason for this is that with edge-to-edge, there's
+        // a larger map canvas available.
+        val endPaddingDp =
+            windowInsetPadding.calculateEndPadding(layoutDirection) +
+                gridPadding.calculateEndPadding(layoutDirection)
+        val endOffsetDp = if (isLandscape) endPaddingDp else horizontalPadding
+
+        val bottomPaddingDp =
+            windowInsetPadding.calculateBottomPadding() + gridPadding.calculateBottomPadding()
+        val bottomOffsetDp =
+            if (isLandscape) bottomPaddingDp else bottomPaddingDp + arrivalViewHeight
+
+        // TODO: This could be improved if we want to add pixel width to dp conversion in
+        //  maplibre-compose.
+        val attributionOffset = 24.dp
+
+        value =
```

Comment:
Look into Android Studio lint 🤔

Reply:
https://issuetracker.google.com/issues/349411310 added to the comments. Seems like they're messing with this lint  rule and we should be able to eventually remove our suppress.

Reply:
Thanks for digging that up and commenting the code. Yeah, looks like a regression.